acra/configuring-maintaining/general-configuration/acra-rollback.md

```diff
@@ -0,0 +1,167 @@
+
+#### Redis
+
+* `--redis_db_keys=<number>`
+
+  Redis database number to use.
+  Default is `0`.
+  <!-- `acra-server -help` says default is `-1` but in `cmd/redis.go` I see `redisDefaultDB = 0` -->
+  <!-- this var is also used as default value for the flag, where's the truth? -->
+
+* `--redis_host_port=<host:port>`
+
+  Address of Redis database to use as keystore.
+  If not specified, Redis is not used.
+
+* `--redis_password=<password>`
+
+  Password to Redis database.
+
+
+### HashiCorp Vault
+
+`acra-addzone` can read `ACRA_MASTER_KEY` from HashiCorp Vault instead of environment variable.
+
+* `--vault_connection_api_string=<url>`
+
+  Connection string (like `https://example.com:8200`) for connecting to HashiCorp Vault.
+  If not specified, `ACRA_MASTER_KEY` environment variable will be used.
+
+* `--vault_secrets_path=<kv-path>`
+
+  Path to KV Secrets directory in Vault used to store `ACRA_MASTER_KEY`.
+  Default is `secret/`.
+
+* `--vault_tls_ca_path=<path>`
+
+  Path to CA certificate bundle to use for HashiCorp Vault certificate validation.
+
+  If not specified, use root certificates configured in system.
+
+* `--vault_tls_client_cert=<path>`
+
+  Path to client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_client_key=<path>`
+
+  Path to the private key of the client TLS certificate used to connect to HashiCorp Vault.
+
+  If not specified, don't send client certificate.
+
+* `--vault_tls_transport_enable={true|false}`
+
+  Use TLS to encrypt transport with HashiCorp Vault.
+  Default is `false`.
+
+## Usage example
+
+{{< hint warning >}}
+**Note:**
+Starting with Acra [`v.0.77.0`](https://github.com/cossacklabs/acra/releases/tag/0.77.0), `acra-rollback` requires Go version >= 1.8.
+{{< /hint >}}
+
+
+Single-quote syntax with $ escaping:
+
+```
+acra-rollback --client_id=client --postgresql_enable --connection_string="dbname=acra user=postgres password=postgres host=127.0.0.1 port=5432" --output_file=out.txt --select="select data from test_example_without_zone;" --insert="insert into test_example_without_zone values(\$1);"
+```
+
+
+Double-quote syntax::
+
+```
+acra-rollback --client_id=client --postgresql_enable --connection_string="dbname=acra user=postgres password=postgres host=127.0.0.1 port=5432" --output_file=out.txt --select="select data from test_example_without_zone;" --insert='insert into test_example_without_zone values($1);'
+```
+
+### ZoneMode
+
+`acra-rollback` support work with [zones]({{< ref "/acra/security-controls/zones.md" >}}), you can configure it via `zonemode_enable` flag.
+If zonemode is enabled, make sure you have Zone in your SELECT query:
+
+```
+select zone_id, encrypted_data from some_table;
+```
+
+
+
+### Saving decrypted data to file
+
+Instead of inserting data back into the database, you can print it to the output file, to handle it later. To do it, change the insert query to a simple `$1;`, like this:
```

Comment:
```suggestion
Instead of inserting data back into the database, you can write it to the output file, to handle it later. To do it, change the `INSERT` query to a simple `$1;`, like this:
```